### ISSUE TITLE
Wrong result in Core / Intersection2

### ISSUE THREAD

**Issue by laurentbartholdi** [gap-packages/polycyclic#33]
I can produce an infinite pcp group, with a finite-index subgroup, whose core has infinite index!

```
% gap
 ┌───────┐   GAP 4.10dev-1656-ga092263 of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin15.6.0-default64-kv6
 Configuration:  gmp 6.1.2, GASMAN
 Loading the library and packages ...
 Packages:   Alnuth 3.1.0, ANUPQ 3.3dev, AutPGrp 1.10, FGA 1.2.0, GAPDoc 1.6.1, IO 4.5.4, nq 2.5.3,
             Polycyclic 2.14dev, PrimGrp 3.2.0, SmallGrp 1.2, TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> free := FreeGroup(4);
<free group on the generators [ f1, f2, f3, f4 ]>
gap> AssignGeneratorVariables(free);
#I  Assigned the global variables [ f1, f2, f3, f4 ]
gap> q := free / [ f1^2*f3^-1*f2^-2,
>  f1^-1*f2*f1*f3^-1*f2^-1,
>  f3^2,
>  f1^-1*f3*f1*f4^-1*f3^-1,
>  f2^-1*f3*f2*f4^-1*f3^-1,
>  f2*f3*f2^-1*f4^-1*f3^-1,
>  f4^2,
>  f1^-1*f4*f1*f4^-1,
>  f2^-1*f4*f2*f4^-1,
>  f2*f4*f2^-1*f4^-1,
>  f3^-1*f4*f3*f4^-1 ];
<fp group on the generators [ f1, f2, f3, f4 ]>
gap>
gap> List(LowIndexSubgroupsFpGroup(q,4),x->Index(q,Core(q,x)));
[ 1, 2, 8, 2, 8, 2, 3, 4, 8, 8, 4, 4 ]
gap>
gap> pcpq := PcpGroupFpGroupPcPres(q); # also achievable as NilpotentQuotient(q,3);
Pcp-group with orders [ 2, 0, 2, 2 ]
gap>
gap> List(LowIndexSubgroupClassesPcpGroup(pcpq,4),x->Index(pcpq,Representative(x)));
[ 4, 4, 4, 4, 4, 4, 4 ]
gap>
gap> List(LowIndexSubgroupClassesPcpGroup(pcpq,4),x->Index(pcpq,Core(pcpq,Representative(x))));
[ 4, 4, 4, infinity, 8, 8, 8 ]
```

In fact, the problem seems also to appear as follows:
```
gap> sub := Representative(LowIndexSubgroupClassesPcpGroup(pcpq,4)[4]);;
gap> Index(pcpq,Intersection2(sub,sub^pgpq.1));
infinity
```

**Comment by fingolfin** [comment]
Thank you for the report, and sorry for not responding earlier, but I missed it. In any case, I can reproduce the issue.

**Comment by fingolfin** [comment]
So the intersection is wrong:
```
gap> Pcp(sub);
Pcp [ g2, g4 ] with orders [ 0, 2 ]
gap> Pcp(sub^pcpq.1);
Pcp [ g2*g3, g4 ] with orders [ 0, 2 ]
gap> Pcp(Intersection2(sub,sub^pcpq.1));
Pcp [ g4 ] with orders [ 2 ]

# yet we have:
gap> pcpq.2^4 in sub;
true
gap> pcpq.2^4 in sub^pcpq.1;
true
```

**Comment by fingolfin** [comment]
Unfortunately, `NormalIntersection` is not documented and doesn't seem to match the algorithm in Bettina Habil thesis either, and I can't figure it out right now -- perhaps @beick has some time to debug this?

**Comment by fingolfin** [comment]
Actually, I take that back, `NormalIntersection` of course *does* match Bettina's Habil thesis, I don't know what I was thinking/reading back in September...

**Comment by fingolfin** [comment]
@laurentbartholdi I put a fix for this into PR #36. In particular, I now get this for your example:
```
gap> free := FreeGroup(4);
<free group on the generators [ f1, f2, f3, f4 ]>
gap> AssignGeneratorVariables(free);
#I  Assigned the global variables [ f1, f2, f3, f4 ]
gap> q := free / [ f1^2*f3^-1*f2^-2,
>  f1^-1*f2*f1*f3^-1*f2^-1,
>  f3^2,
>  f1^-1*f3*f1*f4^-1*f3^-1,
>  f2^-1*f3*f2*f4^-1*f3^-1,
>  f2*f3*f2^-1*f4^-1*f3^-1,
>  f4^2,
>  f1^-1*f4*f1*f4^-1,
>  f2^-1*f4*f2*f4^-1,
>  f2*f4*f2^-1*f4^-1,
>  f3^-1*f4*f3*f4^-1 ];
<fp group on the generators [ f1, f2, f3, f4 ]>
gap> List(LowIndexSubgroupsFpGroup(q,4),x->Index(q,Core(q,x)));
[ 1, 2, 8, 2, 8, 2, 3, 4, 8, 8, 4, 4 ]
gap> pcpq := PcpGroupFpGroupPcPres(q); # also achievable as NilpotentQuotient(q,3);
Pcp-group with orders [ 2, 0, 2, 2 ]
gap> List(LowIndexSubgroupClassesPcpGroup(pcpq,4),x->Index(pcpq,Representative(x)));
[ 4, 4, 4, 4, 4, 4, 4 ]
gap> List(LowIndexSubgroupClassesPcpGroup(pcpq,4),x->Index(pcpq,Core(pcpq,Representative(x))));
[ 4, 4, 4, 8, 8, 8, 8 ]
```

I expect the fix to be in the next polycyclic release, unless @beick points out an issue with it.
